src/run.py

```diff
@@ -0,0 +1,170 @@
+        patchset = pyhf.PatchSet(json.load(open(signal_path)))
+        signal_patch = patchset[model_point]
+    elif bkgonly_path.exists():
+        background_only = json.load(open(bkgonly_path))
+        signal_patch = None
+    else:
+        json_file = open(directory_name.glob('*.json')[0])
+        background_only = json.load(json_file)
+        signal_patch = None
+
+    return background_only, signal_patch
+
+
+def calculate_CLs(bkgonly_json, signal_patch_json):
+    """
+    Calculate the observed CLs and the expected CLs band from a background only
+    and signal patch.
+
+    Args:
+        bkgonly_json: The JSON for the background only model
+        signal_patch_json: The JSON Patch for the signal model
+
+    Returns:
+        CLs_obs: The observed CLs value
+        CLs_exp: List of the expected CLs value band
+    """
+    workspace = pyhf.workspace.Workspace(bkgonly_json)
+    if signal_patch_json:
+        model = workspace.model(
+            measurement_name=None,
+            patches=[signal_patch_json],
+            modifier_settings={
+                "normsys": {"interpcode": "code4"},
+                "histosys": {"interpcode": "code4p"},
+            },
+        )
+    else:
+        model = workspace.model()
+
+    result = pyhf.infer.hypotest(
+        1.0, workspace.data(model), model, qtilde=True, return_expected_set=True
+    )
+    if isinstance(pyhf.tensorlib, pyhf.tensor.pytorch_backend):
+        return result[0].tolist()[0], result[-1].tolist()
+    else:
+        return result[0].tolist()[0], result[-1].ravel().tolist()
+
+def end(directory_name):
+    shutil.rmtree(directory_name)
+
+
+@click.command()
+@click.option(
+    "-b",
+    "--backend",
+    "backend",
+    help="Name of the pyhf backend to run with.",
+    default="numpy",
+    required=False,
+)
+@click.option(
+    "-p",
+    "--path",
+    "path",
+    help="Local path of workspace.",
+    default=None,
+    required=False,
+)
+@click.option(
+    "-u",
+    "--url",
+    "url",
+    help="Online data link.",
+    default="https://www.hepdata.net/record/resource/1267798?view=true",
+    required=False,
+)
+@click.option(
+    "-m",
+    "--model_point",
+    "model_point",
+    help="Model point.",
+    default=(750, 100),
+    required=True,
+)
+def main(backend, path, url, model_point):
+    """
+    Automatic process of taking pyhf computation.
+    """
+
+    pyhf.set_backend(backend)
+    print(f"Backend set to: {backend}")
+
+    if url:
+        directory_name = downlaod(url)
+    elif path:
+        directory_name = open_local_file(path)
+    else:
+        print("Invalid command!")
+        print("Command help ....")
+        print(
+            "python run.py -b numpy -u https://www.hepdata.net/record/resource/1267798?view=true -m (750, 100)"
+        )
+        print(
+            "python run.py -b numpy -p ../data/1Lbb-likelihoods-hepdata -m (750, 100)"
+        )
+        raise
```

Comment:
Probably should avoid using bare `raise` statements, even if it is for a custom error.

Reply:
Is there any other more elegant way to remind the users of their wrong command?

Reply:
I just meant that using `raise` with no clear indication of what exception you're raising should be avoided. https://docs.python.org/3/tutorial/errors.html#raising-exceptions

I see that you now just `return`, which works too.